Stop using raw pointers in containers in ScriptExecutionContext.h

cdumez opened the pull request:
#### ca9a027f94b8e3e679bed8206ffc15e752464a53
<pre>
Stop using raw pointers in containers in ScriptExecutionContext.h
<a href="https://bugs.webkit.org/show_bug.cgi?id=302764">https://bugs.webkit.org/show_bug.cgi?id=302764</a>

Reviewed by Darin Adler.

To make it happens, I had to make ContextDestructionObserver subclass
CanMakeWeakPtr. Both ActiveDOMObject and ContextDestructionObserver were
used as raw pointers but the good news is that ActiveDOMObject subclasses
ContextDestructionObserver and is thus now CanMakeWeakPtr as well.

Use WeakHashSet in ScriptExecutionContext instead of HashSet of raw pointers,
which is a lot safer.

* Source/WebCore/Modules/WebGPU/GPUDevice.h:
* Source/WebCore/Modules/applepay-ams-ui/ApplePayAMSUIPaymentHandler.h:
* Source/WebCore/Modules/applepay/ApplePaySession.h:
* Source/WebCore/Modules/applepay/paymentrequest/ApplePayPaymentHandler.h:
* Source/WebCore/Modules/encryptedmedia/CDM.h:
* Source/WebCore/Modules/encryptedmedia/MediaKeySystemAccess.h:
* Source/WebCore/Modules/entriesapi/ErrorCallback.h:
* Source/WebCore/Modules/entriesapi/FileCallback.h:
* Source/WebCore/Modules/entriesapi/FileSystemEntriesCallback.h:
* Source/WebCore/Modules/entriesapi/FileSystemEntryCallback.h:
* Source/WebCore/Modules/fetch/FetchBodyOwner.h:
* Source/WebCore/Modules/filesystem/FileSystemSyncAccessHandle.h:
* Source/WebCore/Modules/geolocation/Geolocation.h:
* Source/WebCore/Modules/geolocation/PositionCallback.h:
* Source/WebCore/Modules/geolocation/PositionErrorCallback.h:
* Source/WebCore/Modules/identity/CredentialRequestCoordinator.h:
* Source/WebCore/Modules/indexeddb/IDBRequest.h:
* Source/WebCore/Modules/indexeddb/IDBTransaction.h:
* Source/WebCore/Modules/mediacontrols/MediaControlsUtils.h:
* Source/WebCore/Modules/mediasession/MediaSession.h:
(WebCore::MediaSession::visitActionHandlers const):
* Source/WebCore/Modules/mediasession/MediaSessionActionHandler.h:
* Source/WebCore/Modules/mediasource/MediaSource.h:
* Source/WebCore/Modules/mediasource/SourceBuffer.h:
* Source/WebCore/Modules/mediasource/SourceBufferList.h:
* Source/WebCore/Modules/mediastream/MediaDevices.h:
* Source/WebCore/Modules/mediastream/MediaStreamTrack.h:
* Source/WebCore/Modules/mediastream/MediaStreamTrackProcessor.h:
* Source/WebCore/Modules/mediastream/RTCDTMFSender.h:
* Source/WebCore/Modules/mediastream/RTCDataChannel.cpp:
(WebCore::RTCDataChannel::createMessageQueue):
(WebCore::RTCDataChannel::send):
(WebCore::RTCDataChannel::close):
* Source/WebCore/Modules/mediastream/RTCDataChannel.h:
* Source/WebCore/Modules/mediastream/RTCDtlsTransport.h:
* Source/WebCore/Modules/mediastream/RTCIceTransport.h:
* Source/WebCore/Modules/mediastream/RTCPeerConnection.h:
* Source/WebCore/Modules/mediastream/RTCRtpSFrameTransform.h:
* Source/WebCore/Modules/mediastream/RTCRtpScriptTransformer.h:
* Source/WebCore/Modules/mediastream/RTCSctpTransport.h:
* Source/WebCore/Modules/notifications/Notification.h:
* Source/WebCore/Modules/notifications/NotificationPermissionCallback.h:
* Source/WebCore/Modules/paymentrequest/PaymentRequest.h:
* Source/WebCore/Modules/paymentrequest/PaymentResponse.h:
* Source/WebCore/Modules/pictureinpicture/PictureInPictureWindow.h:
* Source/WebCore/Modules/remoteplayback/RemotePlayback.h:
* Source/WebCore/Modules/remoteplayback/RemotePlaybackAvailabilityCallback.h:
* Source/WebCore/Modules/reporting/ReportingObserverCallback.h:
* Source/WebCore/Modules/reporting/ReportingScope.h:
* Source/WebCore/Modules/screen-wake-lock/WakeLock.h:
* Source/WebCore/Modules/screen-wake-lock/WakeLockSentinel.h:
* Source/WebCore/Modules/speech/SpeechSynthesis.h:
* Source/WebCore/Modules/speech/SpeechSynthesisUtterance.h:
* Source/WebCore/Modules/streams/QueuingStrategySize.h:
* Source/WebCore/Modules/streams/ReadableStream.h:
* Source/WebCore/Modules/streams/StreamPipeToUtilities.cpp:
* Source/WebCore/Modules/streams/StreamTeeUtilities.cpp:
* Source/WebCore/Modules/streams/UnderlyingSourceCancelCallback.h:
* Source/WebCore/Modules/streams/UnderlyingSourcePullCallback.h:
* Source/WebCore/Modules/streams/UnderlyingSourceStartCallback.h:
* Source/WebCore/Modules/web-locks/WebLockGrantedCallback.h:
* Source/WebCore/Modules/web-locks/WebLockManager.h:
* Source/WebCore/Modules/webaudio/AudioBufferCallback.h:
* Source/WebCore/Modules/webaudio/AudioWorkletProcessorConstructor.h:
* Source/WebCore/Modules/webaudio/BaseAudioContext.h:
* Source/WebCore/Modules/webcodecs/WebCodecsAudioData.h:
* Source/WebCore/Modules/webcodecs/WebCodecsAudioDataOutputCallback.h:
* Source/WebCore/Modules/webcodecs/WebCodecsBase.h:
* Source/WebCore/Modules/webcodecs/WebCodecsEncodedAudioChunkOutputCallback.h:
* Source/WebCore/Modules/webcodecs/WebCodecsEncodedVideoChunkOutputCallback.h:
* Source/WebCore/Modules/webcodecs/WebCodecsErrorCallback.h:
* Source/WebCore/Modules/webcodecs/WebCodecsVideoFrame.h:
* Source/WebCore/Modules/webcodecs/WebCodecsVideoFrameOutputCallback.h:
* Source/WebCore/Modules/webdatabase/DatabaseCallback.h:
* Source/WebCore/Modules/webdatabase/SQLStatementCallback.h:
* Source/WebCore/Modules/webdatabase/SQLStatementErrorCallback.h:
* Source/WebCore/Modules/webdatabase/SQLTransactionCallback.h:
* Source/WebCore/Modules/webdatabase/SQLTransactionErrorCallback.h:
* Source/WebCore/Modules/websockets/WebSocket.h:
* Source/WebCore/Modules/webxr/WebXRInputSpace.h:
* Source/WebCore/Modules/webxr/WebXRJointSpace.h:
* Source/WebCore/Modules/webxr/WebXRLayer.h:
* Source/WebCore/Modules/webxr/WebXRReferenceSpace.h:
* Source/WebCore/Modules/webxr/WebXRSpace.h:
* Source/WebCore/Modules/webxr/WebXRSystem.h:
* Source/WebCore/Modules/webxr/XRFrameRequestCallback.h:
* Source/WebCore/animation/CustomEffectCallback.h:
* Source/WebCore/animation/WebAnimation.h:
* Source/WebCore/bindings/js/JSCustomElementInterface.h:
* Source/WebCore/bindings/js/JSDOMGuardedObject.h:
* Source/WebCore/crypto/SubtleCrypto.h:
* Source/WebCore/css/CSSPaintCallback.h:
* Source/WebCore/css/FontFace.h:
* Source/WebCore/css/FontFaceSet.h:
* Source/WebCore/css/MediaQueryList.h:
* Source/WebCore/dom/AbortAlgorithm.h:
* Source/WebCore/dom/AbortSignal.h:
* Source/WebCore/dom/ActiveDOMObject.h:
* Source/WebCore/dom/BroadcastChannel.h:
* Source/WebCore/dom/ContextDestructionObserver.h:
* Source/WebCore/dom/CreateHTMLCallback.h:
* Source/WebCore/dom/CreateScriptCallback.h:
* Source/WebCore/dom/CreateScriptURLCallback.h:
* Source/WebCore/dom/CustomElementRegistry.h:
* Source/WebCore/dom/DataTransferItemList.h:
* Source/WebCore/dom/EventTargetConcrete.h:
* Source/WebCore/dom/IdleRequestCallback.h:
* Source/WebCore/dom/MapperCallback.h:
* Source/WebCore/dom/MessagePort.cpp:
(WebCore::MessagePort::MessagePort):
* Source/WebCore/dom/MessagePort.h:
* Source/WebCore/dom/MutationCallback.h:
* Source/WebCore/dom/NodeFilter.h:
* Source/WebCore/dom/ObservableInspectorAbortCallback.h:
* Source/WebCore/dom/PredicateCallback.h:
* Source/WebCore/dom/ReducerCallback.h:
* Source/WebCore/dom/RequestAnimationFrameCallback.h:
* Source/WebCore/dom/ScriptExecutionContext.cpp:
(WebCore::ScriptExecutionContext::checkConsistency const):
(WebCore::ScriptExecutionContext::~ScriptExecutionContext):
(WebCore::ScriptExecutionContext::forEachActiveDOMObject const):
(WebCore::ScriptExecutionContext::suspendActiveDOMObjectIfNeeded):
(WebCore::ScriptExecutionContext::didCreateActiveDOMObject):
(WebCore::ScriptExecutionContext::willDestroyActiveDOMObject):
(WebCore::ScriptExecutionContext::didCreateDestructionObserver):
(WebCore::ScriptExecutionContext::willDestroyDestructionObserver):
(WebCore::ScriptExecutionContext::hasPendingActivity const):
* Source/WebCore/dom/ScriptExecutionContext.h:
* Source/WebCore/dom/StringCallback.h:
* Source/WebCore/dom/Subscriber.h:
* Source/WebCore/dom/SubscriberCallback.h:
* Source/WebCore/dom/SubscriptionObserverCallback.h:
* Source/WebCore/dom/TrustedTypePolicyFactory.h:
* Source/WebCore/dom/ViewTransition.h:
* Source/WebCore/dom/ViewTransitionUpdateCallback.h:
* Source/WebCore/dom/VisitorCallback.h:
* Source/WebCore/fileapi/BlobCallback.h:
* Source/WebCore/fileapi/FileReader.h:
* Source/WebCore/fileapi/NetworkSendQueue.cpp:
(WebCore::NetworkSendQueue::create):
(WebCore::NetworkSendQueue::enqueue):
* Source/WebCore/fileapi/NetworkSendQueue.h:
* Source/WebCore/html/DOMFormData.h:
* Source/WebCore/html/HTMLImageElement.h:
* Source/WebCore/html/HTMLSourceElement.h:
* Source/WebCore/html/ImageBitmap.cpp:
* Source/WebCore/html/MediaController.h:
* Source/WebCore/html/OffscreenCanvas.h:
* Source/WebCore/html/VideoFrameRequestCallback.h:
* Source/WebCore/html/VoidCallback.h:
* Source/WebCore/html/canvas/GPUBasedCanvasRenderingContext.h:
* Source/WebCore/html/canvas/WebGLProgram.h:
* Source/WebCore/html/closewatcher/CloseWatcher.h:
* Source/WebCore/html/track/TextTrack.h:
* Source/WebCore/html/track/TextTrackCue.h:
* Source/WebCore/html/track/TrackBase.h:
* Source/WebCore/html/track/TrackListBase.h:
* Source/WebCore/html/track/VTTRegion.h:
* Source/WebCore/inspector/RTCLogsCallback.h:
* Source/WebCore/loader/MediaResourceLoader.h:
* Source/WebCore/page/Crypto.h:
* Source/WebCore/page/DOMTimer.h:
* Source/WebCore/page/IntersectionObserverCallback.h:
* Source/WebCore/page/LocalDOMWindow.h:
* Source/WebCore/page/NavigationHistoryEntry.h:
* Source/WebCore/page/NavigationInterceptHandler.h:
* Source/WebCore/page/NavigatorBase.h:
* Source/WebCore/page/Performance.h:
* Source/WebCore/page/PerformanceObserverCallback.h:
* Source/WebCore/page/ResizeObserverCallback.h:
* Source/WebCore/testing/EventTargetForTesting.h:
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/XRSimulateUserActivationFunction.h:
* Source/WebCore/workers/WorkerEventLoop.h:
* Source/WebCore/workers/service/ServiceWorker.h:
* Source/WebCore/workers/service/ServiceWorkerClient.h:
* Source/WebCore/workers/service/ServiceWorkerContainer.h:
* Source/WebCore/workers/service/ServiceWorkerRegistration.h:
* Source/WebCore/workers/shared/SharedWorker.h:
* Source/WebCore/worklets/Worklet.h:
* Source/WebCore/xml/CustomXPathNSResolver.h:
* Source/WebKit/WebProcess/InjectedBundle/DOM/InjectedBundleNodeHandle.h:
* Source/WebKit/WebProcess/Network/WebSocketChannel.cpp:
(WebKit::WebSocketChannel::createMessageQueue):
(WebKit::WebSocketChannel::send):
(WebKit::WebSocketChannel::disconnect):
* Source/WebKit/WebProcess/Network/WebSocketChannel.h:

Canonical link: <a href="https://commits.webkit.org/303309@main">https://commits.webkit.org/303309@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2c7467b33297faa1aeeb1ba618211bbe75e9eb65

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/131987 "6 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/4479 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/43002 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/139501 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/83888 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/1bd5ab0d-1163-449b-a349-294b6c699ef9) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/133857 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/4422 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/4241 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/100890 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/83888 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/56103313-0f4d-42b9-95d8-7b556c576b9a) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/134933 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/3150 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/118206 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/81681 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/d1e6089b-793b-444c-b1ec-b8ea440b9a05) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/3044 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/911 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/82719 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/111797 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/36327 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/142146 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/4149 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/36909 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/109262 "Found 1 new test failure: imported/blink/fast/pagination/viewport-y-horizontal-bt-rtl.html (failure)") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/4230 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/161/builds/3617 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/109433 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27717 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/3141 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/114485 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/57370 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/4202 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/32881 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/4034 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/67649 "Built successfully") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/163/builds/4294 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/4162 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/Modules/mediastream/MediaStreamTrack.h:
```diff
@@ -65,6 +65,7 @@ class MediaStreamTrack
 {
     WTF_MAKE_TZONE_OR_ISO_ALLOCATED(MediaStreamTrack);
 public:
+    USING_CAN_MAKE_WEAKPTR(EventTarget);
```
This pull request includes some examples with a blank line between `USING_CAN_MAKE_WEAKPTR`  and `ref`/`deref` and others without. I would prefer consistency, although I don’t have preference for which style. Also, in some places `USING_CAN_MAKE_WEAKPTR` is after the `ref`/`deref` functions, another unnecessary inconsistency.
```suggestion
    USING_CAN_MAKE_WEAKPTR(EventTarget);

```

darinadler reviewed (approved):
Does this make any objects larger? Is there a memory use benchmark that will notice memory costs of this kind of change?

cdumez committed:
Stop using raw pointers in containers in ScriptExecutionContext.h
https://bugs.webkit.org/show_bug.cgi?id=302764

Reviewed by Darin Adler.

To make it happens, I had to make ContextDestructionObserver subclass
CanMakeWeakPtr. Both ActiveDOMObject and ContextDestructionObserver were
used as raw pointers but the good news is that ActiveDOMObject subclasses
ContextDestructionObserver and is thus now CanMakeWeakPtr as well.

Use WeakHashSet in ScriptExecutionContext instead of HashSet of raw pointers,
which is a lot safer.

* Source/WebCore/Modules/WebGPU/GPUDevice.h:
* Source/WebCore/Modules/applepay-ams-ui/ApplePayAMSUIPaymentHandler.h:
* Source/WebCore/Modules/applepay/ApplePaySession.h:
* Source/WebCore/Modules/applepay/paymentrequest/ApplePayPaymentHandler.h:
* Source/WebCore/Modules/encryptedmedia/CDM.h:
* Source/WebCore/Modules/encryptedmedia/MediaKeySystemAccess.h:
* Source/WebCore/Modules/entriesapi/ErrorCallback.h:
* Source/WebCore/Modules/entriesapi/FileCallback.h:
* Source/WebCore/Modules/entriesapi/FileSystemEntriesCallback.h:
* Source/WebCore/Modules/entriesapi/FileSystemEntryCallback.h:
* Source/WebCore/Modules/fetch/FetchBodyOwner.h:
* Source/WebCore/Modules/filesystem/FileSystemSyncAccessHandle.h:
* Source/WebCore/Modules/geolocation/Geolocation.h:
* Source/WebCore/Modules/geolocation/PositionCallback.h:
* Source/WebCore/Modules/geolocation/PositionErrorCallback.h:
* Source/WebCore/Modules/identity/CredentialRequestCoordinator.h:
* Source/WebCore/Modules/indexeddb/IDBRequest.h:
* Source/WebCore/Modules/indexeddb/IDBTransaction.h:
* Source/WebCore/Modules/mediacontrols/MediaControlsUtils.h:
* Source/WebCore/Modules/mediasession/MediaSession.h:
(WebCore::MediaSession::visitActionHandlers const):
* Source/WebCore/Modules/mediasession/MediaSessionActionHandler.h:
* Source/WebCore/Modules/mediasource/MediaSource.h:
* Source/WebCore/Modules/mediasource/SourceBuffer.h:
* Source/WebCore/Modules/mediasource/SourceBufferList.h:
* Source/WebCore/Modules/mediastream/MediaDevices.h:
* Source/WebCore/Modules/mediastream/MediaStreamTrack.h:
* Source/WebCore/Modules/mediastream/MediaStreamTrackProcessor.h:
* Source/WebCore/Modules/mediastream/RTCDTMFSender.h:
* Source/WebCore/Modules/mediastream/RTCDataChannel.cpp:
(WebCore::RTCDataChannel::createMessageQueue):
(WebCore::RTCDataChannel::send):
(WebCore::RTCDataChannel::close):
* Source/WebCore/Modules/mediastream/RTCDataChannel.h:
* Source/WebCore/Modules/mediastream/RTCDtlsTransport.h:
* Source/WebCore/Modules/mediastream/RTCIceTransport.h:
* Source/WebCore/Modules/mediastream/RTCPeerConnection.h:
* Source/WebCore/Modules/mediastream/RTCRtpSFrameTransform.h:
* Source/WebCore/Modules/mediastream/RTCRtpScriptTransformer.h:
* Source/WebCore/Modules/mediastream/RTCSctpTransport.h:
* Source/WebCore/Modules/notifications/Notification.h:
* Source/WebCore/Modules/notifications/NotificationPermissionCallback.h:
* Source/WebCore/Modules/paymentrequest/PaymentRequest.h:
* Source/WebCore/Modules/paymentrequest/PaymentResponse.h:
* Source/WebCore/Modules/pictureinpicture/PictureInPictureWindow.h:
* Source/WebCore/Modules/remoteplayback/RemotePlayback.h:
* Source/WebCore/Modules/remoteplayback/RemotePlaybackAvailabilityCallback.h:
* Source/WebCore/Modules/reporting/ReportingObserverCallback.h:
* Source/WebCore/Modules/reporting/ReportingScope.h:
* Source/WebCore/Modules/screen-wake-lock/WakeLock.h:
* Source/WebCore/Modules/screen-wake-lock/WakeLockSentinel.h:
* Source/WebCore/Modules/speech/SpeechSynthesis.h:
* Source/WebCore/Modules/speech/SpeechSynthesisUtterance.h:
* Source/WebCore/Modules/streams/QueuingStrategySize.h:
* Source/WebCore/Modules/streams/ReadableStream.h:
* Source/WebCore/Modules/streams/StreamPipeToUtilities.cpp:
* Source/WebCore/Modules/streams/StreamTeeUtilities.cpp:
* Source/WebCore/Modules/streams/UnderlyingSourceCancelCallback.h:
* Source/WebCore/Modules/streams/UnderlyingSourcePullCallback.h:
* Source/WebCore/Modules/streams/UnderlyingSourceStartCallback.h:
* Source/WebCore/Modules/web-locks/WebLockGrantedCallback.h:
* Source/WebCore/Modules/web-locks/WebLockManager.h:
* Source/WebCore/Modules/webaudio/AudioBufferCallback.h:
* Source/WebCore/Modules/webaudio/AudioWorkletProcessorConstructor.h:
* Source/WebCore/Modules/webaudio/BaseAudioContext.h:
* Source/WebCore/Modules/webcodecs/WebCodecsAudioData.h:
* Source/WebCore/Modules/webcodecs/WebCodecsAudioDataOutputCallback.h:
* Source/WebCore/Modules/webcodecs/WebCodecsBase.h:
* Source/WebCore/Modules/webcodecs/WebCodecsEncodedAudioChunkOutputCallback.h:
* Source/WebCore/Modules/webcodecs/WebCodecsEncodedVideoChunkOutputCallback.h:
* Source/WebCore/Modules/webcodecs/WebCodecsErrorCallback.h:
* Source/WebCore/Modules/webcodecs/WebCodecsVideoFrame.h:
* Source/WebCore/Modules/webcodecs/WebCodecsVideoFrameOutputCallback.h:
* Source/WebCore/Modules/webdatabase/DatabaseCallback.h:
* Source/WebCore/Modules/webdatabase/SQLStatementCallback.h:
* Source/WebCore/Modules/webdatabase/SQLStatementErrorCallback.h:
* Source/WebCore/Modules/webdatabase/SQLTransactionCallback.h:
* Source/WebCore/Modules/webdatabase/SQLTransactionErrorCallback.h:
* Source/WebCore/Modules/websockets/WebSocket.h:
* Source/WebCore/Modules/webxr/WebXRInputSpace.h:
* Source/WebCore/Modules/webxr/WebXRJointSpace.h:
* Source/WebCore/Modules/webxr/WebXRLayer.h:
* Source/WebCore/Modules/webxr/WebXRReferenceSpace.h:
* Source/WebCore/Modules/webxr/WebXRSpace.h:
* Source/WebCore/Modules/webxr/WebXRSystem.h:
* Source/WebCore/Modules/webxr/XRFrameRequestCallback.h:
* Source/WebCore/animation/CustomEffectCallback.h:
* Source/WebCore/animation/WebAnimation.h:
* Source/WebCore/bindings/js/JSCustomElementInterface.h:
* Source/WebCore/bindings/js/JSDOMGuardedObject.h:
* Source/WebCore/crypto/SubtleCrypto.h:
* Source/WebCore/css/CSSPaintCallback.h:
* Source/WebCore/css/FontFace.h:
* Source/WebCore/css/FontFaceSet.h:
* Source/WebCore/css/MediaQueryList.h:
* Source/WebCore/dom/AbortAlgorithm.h:
* Source/WebCore/dom/AbortSignal.h:
* Source/WebCore/dom/ActiveDOMObject.h:
* Source/WebCore/dom/BroadcastChannel.h:
* Source/WebCore/dom/ContextDestructionObserver.h:
* Source/WebCore/dom/CreateHTMLCallback.h:
* Source/WebCore/dom/CreateScriptCallback.h:
* Source/WebCore/dom/CreateScriptURLCallback.h:
* Source/WebCore/dom/CustomElementRegistry.h:
* Source/WebCore/dom/DataTransferItemList.h:
* Source/WebCore/dom/EventTargetConcrete.h:
* Source/WebCore/dom/IdleRequestCallback.h:
* Source/WebCore/dom/MapperCallback.h:
* Source/WebCore/dom/MessagePort.cpp:
(WebCore::MessagePort::MessagePort):
* Source/WebCore/dom/MessagePort.h:
* Source/WebCore/dom/MutationCallback.h:
* Source/WebCore/dom/NodeFilter.h:
* Source/WebCore/dom/ObservableInspectorAbortCallback.h:
* Source/WebCore/dom/PredicateCallback.h:
* Source/WebCore/dom/ReducerCallback.h:
* Source/WebCore/dom/RequestAnimationFrameCallback.h:
* Source/WebCore/dom/ScriptExecutionContext.cpp:
(WebCore::ScriptExecutionContext::checkConsistency const):
(WebCore::ScriptExecutionContext::~ScriptExecutionContext):
(WebCore::ScriptExecutionContext::forEachActiveDOMObject const):
(WebCore::ScriptExecutionContext::suspendActiveDOMObjectIfNeeded):
(WebCore::ScriptExecutionContext::didCreateActiveDOMObject):
(WebCore::ScriptExecutionContext::willDestroyActiveDOMObject):
(WebCore::ScriptExecutionContext::didCreateDestructionObserver):
(WebCore::ScriptExecutionContext::willDestroyDestructionObserver):
(WebCore::ScriptExecutionContext::hasPendingActivity const):
* Source/WebCore/dom/ScriptExecutionContext.h:
* Source/WebCore/dom/StringCallback.h:
* Source/WebCore/dom/Subscriber.h:
* Source/WebCore/dom/SubscriberCallback.h:
* Source/WebCore/dom/SubscriptionObserverCallback.h:
* Source/WebCore/dom/TrustedTypePolicyFactory.h:
* Source/WebCore/dom/ViewTransition.h:
* Source/WebCore/dom/ViewTransitionUpdateCallback.h:
* Source/WebCore/dom/VisitorCallback.h:
* Source/WebCore/fileapi/BlobCallback.h:
* Source/WebCore/fileapi/FileReader.h:
* Source/WebCore/fileapi/NetworkSendQueue.cpp:
(WebCore::NetworkSendQueue::create):
(WebCore::NetworkSendQueue::enqueue):
* Source/WebCore/fileapi/NetworkSendQueue.h:
* Source/WebCore/html/DOMFormData.h:
* Source/WebCore/html/HTMLImageElement.h:
* Source/WebCore/html/HTMLSourceElement.h:
* Source/WebCore/html/ImageBitmap.cpp:
* Source/WebCore/html/MediaController.h:
* Source/WebCore/html/OffscreenCanvas.h:
* Source/WebCore/html/VideoFrameRequestCallback.h:
* Source/WebCore/html/VoidCallback.h:
* Source/WebCore/html/canvas/GPUBasedCanvasRenderingContext.h:
* Source/WebCore/html/canvas/WebGLProgram.h:
* Source/WebCore/html/closewatcher/CloseWatcher.h:
* Source/WebCore/html/track/TextTrack.h:
* Source/WebCore/html/track/TextTrackCue.h:
* Source/WebCore/html/track/TrackBase.h:
* Source/WebCore/html/track/TrackListBase.h:
* Source/WebCore/html/track/VTTRegion.h:
* Source/WebCore/inspector/RTCLogsCallback.h:
* Source/WebCore/loader/MediaResourceLoader.h:
* Source/WebCore/page/Crypto.h:
* Source/WebCore/page/DOMTimer.h:
* Source/WebCore/page/IntersectionObserverCallback.h:
* Source/WebCore/page/LocalDOMWindow.h:
* Source/WebCore/page/NavigationHistoryEntry.h:
* Source/WebCore/page/NavigationInterceptHandler.h:
* Source/WebCore/page/NavigatorBase.h:
* Source/WebCore/page/Performance.h:
* Source/WebCore/page/PerformanceObserverCallback.h:
* Source/WebCore/page/ResizeObserverCallback.h:
* Source/WebCore/testing/EventTargetForTesting.h:
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/XRSimulateUserActivationFunction.h:
* Source/WebCore/workers/WorkerEventLoop.h:
* Source/WebCore/workers/service/ServiceWorker.h:
* Source/WebCore/workers/service/ServiceWorkerClient.h:
* Source/WebCore/workers/service/ServiceWorkerContainer.h:
* Source/WebCore/workers/service/ServiceWorkerRegistration.h:
* Source/WebCore/workers/shared/SharedWorker.h:
* Source/WebCore/worklets/Worklet.h:
* Source/WebCore/xml/CustomXPathNSResolver.h:
* Source/WebKit/WebProcess/InjectedBundle/DOM/InjectedBundleNodeHandle.h:
* Source/WebKit/WebProcess/Network/WebSocketChannel.cpp:
(WebKit::WebSocketChannel::createMessageQueue):
(WebKit::WebSocketChannel::send):
(WebKit::WebSocketChannel::disconnect):
* Source/WebKit/WebProcess/Network/WebSocketChannel.h:

Canonical link: https://commits.webkit.org/303309@main
